mitodl/micromasters
# Sent emails when new user fills out profile and their profile matches query

#### What are the relevant tickets?
Fixes #2705

#### What's this PR do?
When a new user fills out their profile and one of the percolated queries matches their information, they should get an email based on the contents of the relevant `AutomaticEmail` model. There should also be one `SentAutomaticEmail` created for each user for each `AutomaticEmail` after the email is successfully sent.

#### How should this be manually tested?
Edit `static/js/lib/api.js` to add `send_automatic_emails: true` to the request parameters for the search API. Then go to `/learners` and do a search. Send an email. You should see a new `AutomaticEmail` model and a new `SentAutomaticEmail` entry for each user, for that `AutomaticEmail` model, and you should receive an email with a list of addresses matching the profiles in your search where `email_optin` was true.

Next, choose one of those profiles. Delete the `SentAutomaticEmail` entry. Update its employment information somehow (either log in as the user, or modify its model directly). After the update goes though you should receive a new email and there should be a new entry for `SentAutomaticEmail` replacing the one you deleted earlier.

Next, modify the education for the profile. Make sure that you did not receive an email in response to this change.
